libres/lib/job_queue/ext_job.cpp

```diff
@@ -1020,12 +1019,20 @@ ext_job_type * ext_job_fscanf_alloc(const char * name , const char * license_roo
           ext_job_iset_argtype_string( ext_job , iarg , arg_type );
         }
 
-        {
-          const char * executable     = config_content_get_value_as_executable(content  , EXECUTABLE_KEY);
-          const char * executable_raw = config_content_iget(content  , EXECUTABLE_KEY , 0,0);
-          ext_job_set_executable(ext_job , executable, executable_raw, search_path);
+        char exec_key[20] = EXECUTABLE_KEY;
+        if (!config_content_has_item(content , EXECUTABLE_KEY) && !config_content_has_item(content , "PORTABLE_EXE")) {
+            fprintf(stderr, "%s: ** %s must be set\n", config_file, EXECUTABLE_KEY);
+            ext_job->__valid = false;
+        } else if (!config_content_has_item(content , EXECUTABLE_KEY) && config_content_has_item(content , "PORTABLE_EXE")) {
+            fprintf(stderr, "%s: PORTABLE_EXE is deprecated. Use %s instead.\n", config_file, EXECUTABLE_KEY);
+            strcpy(exec_key, "PORTABLE_EXE");
         }
 
+        if (ext_job->__valid) {
+            const char * executable     = config_content_get_value_as_executable(content, exec_key);
+            const char * executable_raw = config_content_iget(content, exec_key, 0, 0);
+            ext_job_set_executable(ext_job , executable, executable_raw, search_path);
+        }
```

Comment:
Should we have a warning if both are set as well (is this even allowed??)? Right now using both will mean that EXECUTABLE_KEY will take priority (which is good), but it also silences the warning about PORTABLE_EXE? :thinking:

Reply:
Good catch!

Reply:
I've made it so that we at least explain what we've done (ignored PORTABLE_EXE).

Reply:
And introduced general warning output in `ext_job`.